### ISSUE TITLE
TextView: Setting `Text` places cursor at beginning, unlike `TextField`

### ISSUE THREAD

**Issue by tig** [gui-cs/Terminal.Gui#2196]
TextView always sets currentRow/currentCol to 0 when `Text` is set.

Note `ResetPosition` call:

```cs
		public override ustring Text {
			get {
				if (wordWrap) {
					return wrapManager.Model.ToString ();
				} else {
					return model.ToString ();
				}
			}

			set {
				ResetPosition ();
				model.LoadString (value);
				if (wordWrap) {
					wrapManager = new WordWrapManager (model);
					model = wrapManager.WrapModel (frameWidth, out _, out _, out _, out _);
				}
				TextChanged?.Invoke ();
				SetNeedsDisplay ();

				historyText.Clear (Text);
			}
		}
```

TextField is more intelligent (`point`) is the cursor pos:

```cs
		public new ustring Text {
			get {
				return ustring.Make (text);
			}

			set {
				var oldText = ustring.Make (text);

				if (oldText == value)
					return;

				var newText = OnTextChanging (value.Replace ("\t", "").Split ("\n") [0]);
				if (newText.Cancel) {
					if (point > text.Count) {
						point = text.Count;
					}
					return;
				}
				text = TextModel.ToRunes (newText.NewText);

				if (!Secret && !historyText.IsFromHistory) {
					historyText.Add (new List<List<Rune>> () { oldText.ToRuneList () },
						new Point (point, 0));
					historyText.Add (new List<List<Rune>> () { text }, new Point (point, 0)
						, HistoryText.LineStatus.Replaced);
				}

				TextChanged?.Invoke (oldText);

				if (point > text.Count) {
					point = Math.Max (TextModel.DisplaySize (text, 0).size - 1, 0);
				}

				Adjust ();
				SetNeedsDisplay ();
			}
		}
```

Debatable if this is broken. I discovered it writing unit tests for fixing #730.

**Comment by BDisp** [comment]
Normally big text editors, as `TextView` is, place the cursor at the beginning to avoid navigate unnecessarily to the end and also if the user want to make some search will start find forward which is the default. Now, if `MultiLine` is disabled it can acts like the `TextField`.

**Comment by BDisp** [comment]
Fixed with #2572.
